code-challenges.test.js

```diff
@@ -14,39 +16,238 @@
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+// describe("matchTheLetter", () => {
+//     it("creates a function that takes an array of words/single letter and returns an array of all the words with that single letter", () => {
+//     expect(matchTheLetter(filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+//     expect(matchTheLetter(filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//  })
+// })
+
+//RED TEST- Good Failure
+// FAIL  ./code-challenges.test.js
+// matchTheLetter
+//   ✕ creates a function that takes an array of words/single letter and returns an array of all the words with that single letter
+
+// ● matchTheLetter › creates a function that takes an array of words/single letter and returns an array of all the words with that single letter
+
+//   ReferenceError: matchTheLetter is not defined
+
+//     119 | describe("matchTheLetter", () => {
+//     120 |     it("creates a function that takes an array of words/single letter and returns an array of all the words with that single letter", () => {
+//   > 121 |     expect(matchTheLetter(filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+//         |     ^
+//     122 |     expect(matchTheLetter(filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//     123 |  })
+//     124 | })
+
+//     at Object.expect (code-challenges.test.js:121:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.16 s, estimated 1 s
+// Ran all test suites.
+
+
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
+//Pseudo Code:
+//Input: 
+            //const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+            // const filterLetterA = "a"
+            // const filterLetterE = "e"
+//Output:["Mango", "Apricot", "Peach"], ["Cherry", "Blueberry", "Peach"]
+//Process: Create a function called fruitList, pass in the parameters of array and string
+//return - use .filter method on the array to filter through values at the index, pasing in the string parameter
+//create a new variable that takes the function fruitList and passes in the arguements of the variable fruitArray and filterLetterA?? CHATGPT suggested this because I got stuck on how to return it without getting an "undefined". It also didn't work.
+//console.log the function with the passed in arguement of filterLetterA
+//OUTPUT: 'array.fitler is not a function' OR 'fruitList is undefined'
+
+// const fruitList = (array, string) => {
+//     return array.filter(array => array.includes(string))
+//     //const selectedWords = fruitList(fruitArray, filterLetterA);
```

Comment:
You can remove line 162 entirely, and then the function should work just fine. Just remember your test is expecting a function called matchTheLetter, not fruitList. Also, your function is expecting 2 arguments, not just one.